### PR TITLE
Add basic test for deleting users by user role

### LINKED ISSUE #157: Add tests for Deleting Users by User Role
Add tests for following filters

Selections

- User role

Filters

- Restrict to users who are registered in the site for at least x days
- Restrict to users who don't have any posts in post type

Do this for a couple of post types

### PR DESCRIPTION
**Pull request by anandB000**
Fix #157

### CODE REVIEW

File: tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php [hunk 1]
```diff
@@ -109,4 +109,46 @@ public function test_delete_users_by_role_with_filter_set_registered_day_at_leas
...
+			'login_restrict'      => false,
+			'no_posts'            => true,
+			'no_posts_post_types' => 'post'
+		);
+		$this->delete_by_user_role->delete( $delete_options );
+
+		// Assert that user role has one user, $user1 has one post and $user2 has no post.
+		$users_in_role = get_users( array( 'role' => 'subscriber' ) );
+		$author_in_post = get_posts( array( 'post_author' => $user1 ) );
+		$author_in_page = get_posts( array( 'post_author' => $user2, 'post_type' => 'page' ) );
```
Comment (sudar) [inline tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php]: We are testing deletion of users. So check whether `user2` got deleted and `user1` is not deleted in addition to the above checks.

Comment (anandB000) [reply tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php]: Completed.

File: tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		//Create two user and assign to subscriber role
+		$user1 = $this->factory->user->create( array( 'user_login' => 'user_test1', 'user_pass' => 'ZXC987abc', 'role' => 'subscriber' ) );
+		$user2 = $this->factory->user->create( array( 'user_login' => 'user_test2', 'user_pass' => 'ZXC987abc2', 'role' => 'subscriber' ) );
+
+		//Create post and assign author
+		$post = $this->factory->post->create( array( 'post_title' => 'post1', 'post_author' => $user1 ) );
+		$page = $this->factory->post->create( array( 'post_title' => 'page1', 'post_type' => 'page', 'post_author' => $user2 ) );
+
+		// Assert that user role has two users $user1 and $user2.
+		$users_in_role = get_users( array( 'role' => 'subscriber' ) );
```
Comment (sudar) [inline tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php]: You can just use the `assertEquals` function and then pass an array.

The for loop and the subsequent conditions are all not necessary.

Comment (anandB000) [reply tests/wp-unit/tests/include/users/Bulk_Delete_Users_By_User_RoleTest.php]: Foreach loop and the subsequent conditions are removed.

**Pull request merged**